common/src/main/java/com/wynntils/functions/CombatFunctions.java

```diff
@@ -40,4 +42,35 @@ protected List<String> getAliases() {
             return List.of("adavg");
         }
     }
+
+    public static class HeightFunction extends Function<Double> {
+        @Override
+        public Double getValue(FunctionArguments arguments) {
+            // iteratively find the first non-air block below the player
+            double endY = (int) Math.ceil(McUtils.player().position().y) - 1;
```

Comment:
I'm pretty certain Minecraft vanilla keeps a height map where you can look up the highest non-air-block. Don't remember the name of the function though, but google is likely your friend.

Reply:
I originally tried that, but it's not suitable for two reasons:
- It returns an int, so it doesn't work with slabs/carpets which is significant since acrobat wants 1.25 blocks off the ground
- It doesn't play well with caves/verticality at all, there is no way to specify which "pocket" of air you want to look at

Reply:
Okay, but if you have to produce this value yourself, place it in a model. Functions are no such place to do almost any calculation.